specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json

```diff
@@ -0,0 +1,1234 @@
+          "type": "string"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "The properties of the container instance.",
+          "$ref": "#/definitions/ContainerProperties"
+        }
+      }
+    },
+    "ContainerProperties": {
+      "description": "The container instance properties.",
+      "type": "object",
+      "required": [
+        "image",
+        "resources"
+      ],
+      "properties": {
+        "image": {
+          "description": "The name of the image used to create the container instance.",
+          "type": "string"
+        },
+        "command": {
+          "description": "The commands to execute within the container instance in exec form.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ports": {
+          "description": "The exposed ports on the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerPort"
+          }
+        },
+        "environmentVariables": {
+          "description": "The environment variables to set in the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvironmentVariable"
+          }
+        },
+        "instanceView": {
+          "description": "The instance view of the container instance. Only valid in response.",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "restartCount": {
+              "readOnly": true,
+              "type": "integer",
+              "description": "The number of times that the container instance has been restarted."
+            },
+            "currentState": {
+              "readOnly": true,
+              "description": "Current container instance state.",
+              "$ref": "#/definitions/ContainerState"
+            },
+            "previousState": {
+              "readOnly": true,
+              "description": "Previous container instance state.",
+              "$ref": "#/definitions/ContainerState"
+            },
+            "events": {
+              "readOnly": true,
+              "description": "The events of the container instance.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Event"
+              }
+            }
+          }
+        },
+        "resources": {
+          "description": "The resource requirements of the container instance.",
+          "$ref": "#/definitions/ResourceRequirements"
+        },
+        "volumeMounts": {
+          "description": "The volume mounts available to the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VolumeMount"
+          }
+        }
+      }
+    },
+    "ContainerState": {
+      "description": "The container instance state.",
```

Comment:
the name of this type suggest that it is populated only by server - please mark all the properties as read-only

Reply:
Done